src/backgroundFetch.js

```diff
@@ -0,0 +1,167 @@
+import BackgroundFetch from 'react-native-background-fetch'
+import PushNotification from 'react-native-push-notification'
+import { AsyncStorage } from 'react-native'
+import moment from 'moment'
+import Config from './config/config'
+import { IS_LOGGED_IN } from './lib/constants/localStorage'
+import goodWallet from './lib/wallet/GoodWallet'
+import userStorage from './lib/gundb/UserStorage'
+
+const options = {
+  minimumFetchInterval: 15,
+  forceAlarmManager: false,
+  stopOnTerminate: false,
+  startOnBoot: true,
+  requiredNetworkType: BackgroundFetch.NETWORK_TYPE_ANY,
+  requiresCharging: false,
+  requiresDeviceIdle: false,
+  requiresBatteryNotLow: false,
+  requiresStorageNotLow: false,
+  enableHeadless: true,
+  periodic: true,
+}
+
+const task = async taskId => {
+  console.log('[BackgroundFetch] taskId: ', taskId)
+
+  const isLoggedIn = await AsyncStorage.getItem(IS_LOGGED_IN).then(JSON.parse)
+
+  console.log('isLoggedIn', isLoggedIn)
+
+  if (!isLoggedIn) {
+    return BackgroundFetch.finish(taskId)
+  }
+
+  try {
+    await hasConnection()
+  } catch (e) {
+    return BackgroundFetch.finish(taskId)
+  }
+
+  const lastFeedCheck = await userStorage.feed.get('lastSeenDate')
+  const feed = await userStorage.getFeedPage(20, true)
+  const beginningOfDayMillis = moment(lastFeedCheck)
+    .startOf('day')
+    .valueOf()
+
+  console.log('lastFeedCheck', lastFeedCheck)
+  console.log('feed', feed)
+
+  const hasNewPayment = (type, status) => {
+    return type === 'receive' && status === 'completed'
+  }
+
+  const hasNewPaymentWithdraw = (type, status) => {
+    return type === 'send' && status === 'completed'
+  }
+
+  const newFeeds = feed.filter(feedItem => {
+    const dayMillis = moment(feedItem.date)
+      .startOf('day')
+      .valueOf()
+    const { type, status } = feedItem
+    const newFeedItem =
+      (hasNewPayment(type, status) || hasNewPaymentWithdraw(type, status)) && dayMillis >= beginningOfDayMillis
+    return newFeedItem && feedItem
+  })
+
+  await userStorage.feed.put({
+    lastSeenDate: Date.now(),
+  })
+
+  if (!newFeeds) {
+    return BackgroundFetch.finish(taskId)
+  }
+
+  newFeeds.map(feed =>
+    PushNotification.localNotification({
+      title: `${feed.type} operation is ${feed.status}`,
```

Comment:
title: `Payment from/to <recipient name> received/accepted`
message: `G$ <amount>`